examples/Digital filtering/realtime_dwt.py

```diff
@@ -0,0 +1,196 @@
+                t_coeff = np.array(buffer.timestamps)
+                v_coeff = np.array(buffer.values)
+                u_coeff = np.array(buffer.uncertainties)
+
+                # change the scatter
+                data = np.c_[t_coeff, v_coeff]
+                c_line[0].set_offsets(data)
+                c_line[0].set_facecolor(["r"]*len(t_coeff))
+
+                # change the error bars
+                c_line[1].remove()
+                c_line[1] = _ax.errorbar(t_coeff, v_coeff, yerr=u_coeff, linewidth=0, elinewidth=1.5, color="gray", capsize=3, zorder=0)
+                upper_unc = v_coeff+u_coeff
+                lower_unc = v_coeff-u_coeff
+                #data_line, caplines, barlinecols = c_line[1].lines
+                #data_line.set_xdata(t_coeff)
+                #data_line.set_ydata(v_coeff)
+                #caplines[0].set_xdata(t_coeff)
+                #caplines[0].set_ydata(upper_unc)
+                #caplines[1].set_xdata(t_coeff)
+                #caplines[1].set_ydata(lower_unc)
+                
+
+                if v_coeff.size != 0:
+                    lim = [np.min(lower_unc), np.max(upper_unc)]
+                    _ax.set_ylim(lim)
+            
+            # highlight the largest coefficients
+            coeffs_all = [np.array(buffer.values) for i_buffer, buffer in enumerate(output_multi)]
+            coeffs_all_indices = [[(i_buffer, i_coeff) for i_coeff, coeff in enumerate(buffer.values)] for i_buffer, buffer in enumerate(output_multi)]
+            coeffs_joined = list(itertools.chain(*coeffs_all))                      # make 1D list of all coeff-values
+            coeffs_joined_indices = list(itertools.chain(*coeffs_all_indices))      # make 1D list of (i_buffer, i_coeff)
+            coeff_highest = np.argpartition(-np.abs(coeffs_joined), kth=range(10))[:10]  # get 10 biggest (absolut) values
+            coeff_highest_indicies = [coeffs_joined_indices[i] for i in coeff_highest]
+            
+            for coeff_index in coeff_highest_indicies:
+                _ax = ax[1::][::-1][coeff_index[0]]
+                c_line = c_lines[::-1][coeff_index[0]]
+                
+                tmp_colors = c_line[0].get_facecolor()
+                tmp_colors = ["k" if i == coeff_index[1] else c for i, c in enumerate(tmp_colors)]
+                c_line[0].set_facecolor(tmp_colors)
+
+            # finally update the plot itself
+            fig.tight_layout()
+            fig.align_ylabels(ax)
+            fig.canvas.draw()
+            fig.canvas.flush_events()
+
+            # reset plot counter
+            plot_counter = 0
+
+        # stop for plotting
+        #if cycle_counter % 7 == 0 and cycle_counter > buffer_length:
+        #    ex = input("Exit (y/n): ")
+        #    if ex == "y":
+        #        exit()
+
+        #tm.sleep(cycle_duration/2)
+
+
+
+
+
+if __name__ == "__main__":
+    try:
+        main()
+    except KeyboardInterrupt:
+        print("\nExiting.\n")
```

Comment:
Will there be explanation added or will this be transferred into a Jupyter Notebook?

Reply:
:+1: